--- origsrc/jpilot-1.8.1/configure.in	2011-04-05 15:24:39.000000000 -0500
+++ src/jpilot-1.8.1/configure.in	2013-12-23 22:54:00.581300700 -0600
@@ -249,6 +249,11 @@ for pilot_libs in $pilot_prefix/lib /usr
       PILOT_LIBS="-L$pilot_libs $PILOT_LIBS"
       break
    fi
+   if test -r "$pilot_libs/libpisock.dll.a" ; then
+      pilotlibs=yes
+      PILOT_LIBS="-L$pilot_libs $PILOT_LIBS"
+      break
+   fi
 done
 fi
 
