Conflicts with intltool macros called earlier

--- origsrc/jpilot-1.8.2/configure.in	2015-03-19 05:49:36.281738500 -0500
+++ src/jpilot-1.8.2/configure.in	2015-03-19 05:52:50.776436200 -0500
@@ -52,11 +52,6 @@ dnl            which leave this macro un
 DATADIRNAME=share
 AC_SUBST(DATADIRNAME)
 
-dnl Setup gettext using version available on host
-dnl The intl/ directory is not exported with the distribution
-AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.16.1])
-
 if test "x$MSGFMT" = "xno"; then
    if test "x$GMSGFMT" = "xno"; then
       AC_MSG_RESULT(No gettext found)
