--- origsrc/jpilot-1.8.2/Makefile.am	2015-03-19 05:45:44.761839200 -0500
+++ src/jpilot-1.8.2/Makefile.am	2015-03-19 05:46:03.282191000 -0500
@@ -23,7 +23,7 @@ color_DATA = \
 
 # Instructions for the code to build begins here
 
-bin_PROGRAMS = jpilot jpilot-dump jpilot-sync jpilot-merge
+bin_PROGRAMS = jpilot
 
 jpilot_SOURCES = \
 	address.c \
@@ -53,6 +53,9 @@ jpilot_SOURCES = \
 	japanese.h \
 	jpilot.c \
 	jpilot.h \
+	jpilot-dump.c \
+	jpilot-merge.c \
+	jpilot-sync.c \
 	jp-contact.c \
 	jp-pi-contact.h \
 	libplugin.c \
@@ -114,7 +117,6 @@ jpilot_dump_SOURCES = \
 	cp1250.c \
 	datebook.c \
 	japanese.c \
-	jpilot-dump.c \
 	libplugin.c \
 	log.c \
 	memo.c \
@@ -130,7 +132,6 @@ jpilot_dump_SOURCES = \
 jpilot_sync_SOURCES = \
 	cp1250.c \
 	category.c \
-	jpilot-sync.c \
 	japanese.c \
 	libplugin.c \
 	log.c \
@@ -148,7 +149,6 @@ jpilot_merge_SOURCES = \
 	japanese.c \
 	libplugin.c \
 	log.c \
-	jpilot-merge.c \
 	otherconv.c \
 	plugins.c \
 	prefs.c \
@@ -167,7 +167,7 @@ I18NDEFS = -DLOCALEDIR=\"$(localedir)\"
 AM_CFLAGS= @PILOT_FLAGS@ @GTK_CFLAGS@ ${I18NDEFS}
 
 # Add linkflags
-jpilot_LDFLAGS = -export-dynamic
+jpilot_LDFLAGS = -export-dynamic -Wl,--out-implib,libjpilot.dll.a
 jpilot_LDADD=@LIBS@ @PILOT_LIBS@ @GTK_LIBS@
 jpilot_dump_LDADD=@LIBS@ @PILOT_LIBS@ @GTK_LIBS@
 jpilot_sync_LDFLAGS = -export-dynamic
--- origsrc/jpilot-1.8.2/jpilot-dump.c	2014-04-05 17:23:39.000000000 -0500
+++ src/jpilot-1.8.2/jpilot-dump.c	2015-03-19 05:46:03.288691800 -0500
@@ -75,14 +75,12 @@ const char *formatT;
  * The variables below are global variables in jpilot.c which are unused in
  * this code but must be instantiated for the code to compile.  
  * The same is true of the functions which are only used in GUI mode. */
-pid_t jpilot_master_pid = -1;
-int pipe_to_parent;
-GtkWidget *glob_dialog;
-GtkWidget *glob_date_label;
-gint glob_date_timer_tag;
+extern pid_t jpilot_master_pid;
+extern int pipe_to_parent;
+extern GtkWidget *glob_dialog;
+extern GtkWidget *glob_date_label;
+extern gint glob_date_timer_tag;
 
-void output_to_pane(const char *str) { return; }
-int sync_once(struct my_sync_info *sync_info) { return EXIT_SUCCESS; }
 /* End Hack */
 
 /****************************** Main Code *************************************/
@@ -1098,7 +1096,7 @@ static int dumpmemo(void)
    return EXIT_SUCCESS;
 }
 
-int main(int argc, char *argv[])
+int dump_main(int argc, char *argv[])
 {
    int i;
    time_t ltime;
@@ -1128,6 +1126,7 @@ int main(int argc, char *argv[])
    textdomain(EPN);
 #endif
 
+   jpilot_master_pid = -1;	/* used only in GUI mode */
    /* If called with no arguments then print usage information */
    if (argc == 1)
    {
--- origsrc/jpilot-1.8.2/jpilot-merge.c	2014-05-24 22:25:20.000000000 -0500
+++ src/jpilot-1.8.2/jpilot-merge.c	2015-03-19 05:49:00.987756800 -0500
@@ -57,18 +57,9 @@
  * The variables below are global variables in jpilot.c which are unused in
  * this code but must be instantiated for the code to compile.  
  * The same is true of the functions which are only used in GUI mode. */
-pid_t jpilot_master_pid = -1;
-GtkWidget *glob_dialog;
-GtkWidget *glob_date_label;
-gint glob_date_timer_tag;
-
-void output_to_pane(const char *str) {}
-int sync_once(struct my_sync_info *sync_info) { return EXIT_SUCCESS; }
-int jp_pack_Contact(struct Contact *c, pi_buffer_t *buf) { return 0; }
-int edit_cats(GtkWidget *widget, char *db_name, struct CategoryAppInfo *cai)
-{
-   return 0;
-}
+extern pid_t jpilot_master_pid;
+extern GtkWidget *glob_dialog;
+extern GtkWidget *glob_date_label;
 /* End Hack */
 
 /****************************** Main Code *************************************/
@@ -276,7 +267,7 @@ static int merge_pdb_file(char *src_pdb_
 }
 
 
-int main(int argc, char *argv[])
+int merge_main(int argc, char *argv[])
 {
    /* enable internationalization(i18n) before printing any output */
 #if defined(ENABLE_NLS)
@@ -308,6 +299,7 @@ int main(int argc, char *argv[])
    in_pc   = argv[2];
    out_pdb = argv[3];
    
+   jpilot_master_pid = -1;	/* used only in GUI mode */
    merge_pdb_file(in_pdb, in_pc, out_pdb);
    
    return EXIT_SUCCESS;
--- origsrc/jpilot-1.8.2/jpilot-sync.c	2014-05-24 22:25:20.000000000 -0500
+++ src/jpilot-1.8.2/jpilot-sync.c	2015-03-19 05:46:03.294692600 -0500
@@ -37,24 +37,21 @@
 #include "otherconv.h"
 
 /******************************* Global vars **********************************/
-int pipe_to_parent, pipe_from_parent;
-pid_t glob_child_pid;
-unsigned char skip_plugins;
+extern int pipe_to_parent, pipe_from_parent;
+extern pid_t glob_child_pid;
+extern unsigned char skip_plugins;
 
 /* Start Hack */
 /* FIXME: The following is a hack.  
  * The variables below are global variables in jpilot.c which are unused in
  * this code but must be instantiated for the code to compile.  
  * The same is true of the functions which are only used in GUI mode. */
-pid_t jpilot_master_pid = -1;
-int *glob_date_label;
-GtkWidget *glob_dialog;
-GtkTooltips *glob_tooltips;
-gint glob_date_timer_tag;
-
-void output_to_pane(const char *str) { return; }
-void cb_app_button(GtkWidget *widget, gpointer data) { return; }
-void cb_cancel_sync(GtkWidget *widget, unsigned int flags) { return; }
+extern pid_t jpilot_master_pid;
+extern int *glob_date_label;
+extern GtkWidget *glob_dialog;
+extern GtkTooltips *glob_tooltips;
+extern gint glob_date_timer_tag;
+
 /* End Hack */
 
 /****************************** Main Code *************************************/
@@ -97,7 +94,7 @@ static void sig_handler(int sig)
    exit(0);
 }
 
-int main(int argc, char *argv[])
+int sync_main(int argc, char *argv[])
 {
    int flags;
    int r, i;
@@ -119,6 +116,7 @@ int main(int argc, char *argv[])
    textdomain(EPN);
 #endif
 
+   jpilot_master_pid = -1;	/* used only in GUI mode */
    port[0]='\0';
    glob_child_pid=0;
    loop=0;
--- origsrc/jpilot-1.8.2/jpilot.c	2015-03-19 05:45:44.301780800 -0500
+++ src/jpilot-1.8.2/jpilot.c	2015-03-19 05:46:03.298693100 -0500
@@ -121,6 +121,9 @@ extern GtkWidget *monthview_window;
 static void cb_delete_event(GtkWidget *widget, GdkEvent *event, gpointer data);
 static void install_gui_and_size(GtkWidget *main_window);
 
+extern int dump_main(int, char*[]);
+extern int merge_main(int, char*[]);
+extern int sync_main(int, char*[]);
 /****************************** Main Code *************************************/
 
 static int create_main_boxes(void)
@@ -1545,6 +1548,13 @@ int main(int argc, char *argv[])
 #  endif
 #endif
 
+   if (strstr(argv[0], "jpilot-dump"))
+     return dump_main(argc, argv);
+   if (strstr(argv[0], "jpilot-merge"))
+     return merge_main(argc, argv);
+   if (strstr(argv[0], "jpilot-sync"))
+     return sync_main(argc, argv);
+
    skip_plugins=FALSE;
    skip_past_alarms=FALSE;
    skip_all_alarms=FALSE;
